common/src/main/scala/uk/ac/wellcome/sqs/SQSReader.scala

```diff
@@ -39,4 +43,31 @@ class SQSReader @Inject()(sqsClient: AmazonSQS, sqsConfig: SQSConfig)
       .getMessages
       .toList
   }
+
+  private def processAndDeleteMessages(
+    messages: List[Message],
+    process: Message => Unit) = {
+      Future.sequence(messages.map { message =>
+        Future.fromTry(Try(process(message))
+          .recover {
+            case e: Throwable =>
+              error(s"Error processing message", e)
+              throw e
+        }).flatMap(_ => deleteMessage(message))
+      })
+  }
+
+  private def deleteMessage(message: Message) =
+    Future {
+      blocking {
```

Comment:
We're not. As far as I understood is a way of saying to the execution context that the enclosed code is blocking, so that the execution context can deal with it